### ISSUE TITLE
BUG: `vocalpy.segment.ava` occassionally can set last segment > length of sound in samples

### ISSUE THREAD

**Issue by NickleDave** [vocalpy/vocalpy#167]
because we convert from last offset time in seconds, computed from the spectogram time bins, back to sample number, using samplerate

I think this is an edge case that only happens because of vocalpy doing the extra work of converting to sample number for returning a `Segments`.

I determined this was happening with one of the edge cases and looked at the difference -- we give ourselves 408 "extra samples" with this one example. I should confirm this is true for all the other cases that raise the error.

Assuming it's true, we should check for it in a post-condition and fix when it happens.

**Comment by NickleDave** [comment]
After so more detective work, I think I have determined this *is* an edge case, that happens specifically when 
 
By finding a file that throws this error and then manually inspecting inside a call to `vocalpy.ava.segment`, I can see that the last value of `t` itself, as returned by `scipy.signal.stft` is actually greater than the duration of the sound.

```python
((Pdb)) t
array([0.0000000e+00, 2.0480000e-03, 4.0960000e-03, ..., 9.9962880e+00,
       9.9983360e+00, 1.0000384e+01])
((Pdb)) t[-1]
10.000384
((Pdb)) t[-1] * sound.samplerate
2500096.0
((Pdb)) sound.samples
2499585
((Pdb)) sound.duration
9.99834
((Pdb)) offsets[-1] - sound.duration
0.0020439999999997127
```

So when this happens, and the algorithm finds that the last time bin of the spectrogram is a segment offset, then we will already have a segment that is longer than the sound itself, before the function does anything to convert to sample number, etc.

Long story short: edge case, not our fault, we should just catch and fix
